"No locks available." error while boostrapping a YCM superbuild

While configuring the `codyco-superbuild` on the shared disk (NFS) of the new iCubGenova05 setup running Ubuntu 16.04 , I get this error: 

```
 CMake Error at cmake/IncludeUrl.cmake:241 (file):
   error locking file


 "/usr/local/src/robot/codyco-superbuild/build/CMakeFiles/YCMEPHelper.cmake.cma
 ke"

   No locks available.
 Call Stack (most recent call first):
   cmake/YCMBootstrap.cmake:105 (include_url)
   CMakeLists.txt:84 (include)
```

Placing the build in the "normal" filesystem running ext4 solves the problem. 
I guess the error is not directly related to ycm (probably the default configuration for the NFS server in Ubuntu 16.04 are different the lock server is not running by  default) but it is worth opening an issue so future users with the same problem will now how to solve the problem.

--- a/help/release/0.10.0.rst
+++ b/help/release/0.10.0.rst
@@ -31,7 +31,7 @@ Generic Modules
 
 * The :module:`InstallBasicPackageFiles` module learned to forward the
   ``ARCH_INDEPENDENT`` option to :command:`write_basic_package_version_file`.
-* The :module:`IncludeUrl` module now always perform the lock/unlock.
+* The :module:`IncludeUrl` module no longer uses locks.
 
 
 Superbuild Helper Modules

--- a/modules/IncludeUrl.cmake
+++ b/modules/IncludeUrl.cmake
@@ -176,7 +176,6 @@ macro(INCLUDE_URL _remoteFile)
     set(_localFile ${CMAKE_BINARY_DIR}${CMAKE_FILES_DIRECTORY}/${_filename})
     set_property(DIRECTORY APPEND PROPERTY ADDITIONAL_MAKE_CLEAN_FILES "${_localFile}")
   endif()
-  set(_lockFile "${_localFile}.cmake")
 
   if(DEFINED CMAKE_SCRIPT_MODE_FILE)
     string(RANDOM LENGTH 8 _rand)
@@ -232,10 +231,6 @@ macro(INCLUDE_URL _remoteFile)
       set(_expectedHash ${_IU_EXPECTED_MD5})
     endif()
   endif()
-
-  # Lock the file, in case 2 different processes are downloading the same file
-  # at the time
-  file(LOCK "${_lockFile}")
 
   set(_shouldDownload 0)
   set(_shouldFail 0)
@@ -364,9 +359,6 @@ macro(INCLUDE_URL _remoteFile)
       set(${_IU_STATUS} "0")
     endif()
   endif()
-
-  # Download is finished, we can now release the lock
-  file(LOCK "${_lockFile}" RELEASE)
 
   if(NOT EXISTS "${_localFile}" AND NOT _IU_OPTIONAL)
     message(FATAL_ERROR "Downloaded file does not exist. Please report this as a bug")

--- a/modules/YCMEPHelper.cmake
+++ b/modules/YCMEPHelper.cmake
@@ -85,7 +85,7 @@ set(_ycm_CMakeParseArguments_sha1sum 0c4d3f7ed248145cbeb67cbd6fd7190baf2e4517)
 set(_ycm_ExternalProject_sha1sum     62ed9b37172e01fd13f356d8a4a6a599d7aed054)
 
 # Files in all projects that need to bootstrap YCM
-set(_ycm_IncludeUrl_sha1sum          921a037133255d01b644c16f19494cb08d98c462)
+set(_ycm_IncludeUrl_sha1sum          ae6de83398115e5b8fd6c261972dea82d1aeb3fd)
 set(_ycm_YCMBootstrap_sha1sum        dd95e1d38e045091e2e6c1ba2a96d540f1b8af0d)
 
 

